src/payment/strategies/bolt/bolt-payment-strategy.ts

```diff
@@ -144,11 +157,30 @@ export default class BoltPaymentStrategy implements PaymentStrategy {
             throw new MissingDataError(MissingDataErrorType.MissingPayment);
         }
 
-        await this._store.dispatch(this._orderActionCreator.submitOrder(order, _options));
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+
+        const transactionReference = await boltClient.getTransactionReference();
+
+        if (!transactionReference) {
+            return Promise.resolve(this._store.getState());
```

Comment:
Are you sure you want to resolve? I think that'd imply to the client that everything is okay and it can redirect the shopper to the order confirmation page.

Reply:
Oh... My bad. I need to throw an error here to let customer know that something went wrong. I will fix it soon

Reply:
@davidchin Fixed.

I am throwing an error if transactionReference will be empty. In this case, the customer will understand that something went wrong.

Thanks for your comments! 🙏
Now I understand that I need to think more about our customers experience